tests/epyccel/test_epyccel_lists.py

```diff
@@ -841,3 +842,32 @@ def list_str():
     python_result = list_str()
     assert python_result == pyccel_result
 
+def test_list_const_arg(language):
+    @template('T', ['int', 'float', 'complex'])
+    def list_arg(arg : 'const list[T]', my_sum : 'T'):
+        for ai in arg:
+            my_sum += ai
+        return my_sum
+
+    epyccel_func = epyccel(list_arg, language = language)
+    int_arg = [1,2,3,4,5,6,7]
+    float_arg = [1.5, 2.5, 3.5, 4.5, 6.7]
+    complex_arg = [1+0j,4j,2.5+2j]
+    for arg in (int_arg, float_arg, complex_arg):
+        start = type(next(iter(arg)))(0)
+        pyccel_result = epyccel_func(arg, start)
+        python_result = list_arg(arg, start)
+        assert python_result == pyccel_result
+        assert isinstance(pyccel_result, type(python_result))
+
+def test_list_arg(stc_language):
+    def list_arg(arg : 'list[int]', n : int):
```

Comment:
Do we need to write `list[int]` as a string?

Reply:
No. Both in and out of a string are valid and are treated in the same way by Pyccel

Reply:
Then we should test this out of a string for completeness.

BTW, why do you pass `stc_language` here?

Reply:
I am only testing C because Fortran does not accept non-const lists as arguments. It is too costly/complex to retrieve the updated list and copy it through both layers